Editor/EditorCore/EditorUtility.cs

```diff
@@ -292,16 +292,18 @@ internal static void MoveToActiveScene(GameObject gameObject)
         /// Initialize this object with the various editor-only parameters, and invoke the object creation callback.
         /// </summary>
         /// <param name="pb"></param>
-        internal static void InitObject(ProBuilderMesh pb)
+        internal static void InitObject(ProBuilderMesh pb, bool snapToGrid = true)
         {
             MoveToActiveScene(pb.gameObject);
             GameObjectUtility.EnsureUniqueNameForSibling(pb.gameObject);
             ScreenCenter(pb.gameObject);
-            SetPivotLocationAndSnap(pb);
+            
+            if(snapToGrid)
```

Comment:
The name `TrySnapToGrid` implies (to my ear) that it is already doing the snap enabled check. Why is `snapToGrid` also tested?